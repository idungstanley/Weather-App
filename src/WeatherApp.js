import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Home from './Pages/Home';
import Navbar from './Component/Nav/Navbar';
import Check from './Pages/Check';

function WeatherApp() {
  return (
    <main>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/check" element={<Check />} />
        </Routes>
      </Router>
    </main>
  );
}

export default WeatherApp;
