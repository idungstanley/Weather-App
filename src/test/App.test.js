import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import store from './redux/configureStore'
import WeatherApp from "../WeatherApp";

describe("test App components", () => {
  test("renders", () => {
    const tree = render(
      <Provider store={store}>
        <WeatherApp />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
