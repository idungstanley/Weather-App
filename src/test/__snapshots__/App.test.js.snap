// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App components renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="container"
        >
          <img
            alt="logo"
            src="weather-logo.png"
          />
          <ul
            class="link-container"
          >
            <li
              class="link"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
          </ul>
        </div>
        <main>
          <form>
            <input
              class="search"
              id=""
              name=""
              placeholder="Search for Any African country..."
              type="search"
              value=""
            />
          </form>
        </main>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="container"
      >
        <img
          alt="logo"
          src="weather-logo.png"
        />
        <ul
          class="link-container"
        >
          <li
            class="link"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
        </ul>
      </div>
      <main>
        <form>
          <input
            class="search"
            id=""
            name=""
            placeholder="Search for Any African country..."
            type="search"
            value=""
          />
        </form>
      </main>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
